app/jobs/sitemap_monitor_job.rb

```diff
@@ -1,9 +1,10 @@
+# frozen_string_literal: true
+
 class SitemapMonitorJob < ApplicationJob
   queue_as :sitemap
 
   def perform
-    SearchgovDomain.ok.each do |searchgov_domain|
-      searchgov_domain.index_sitemaps
-    end
+    SearchgovDomain.ok.find_each(&:index_sitemaps)
+    SearchgovDomain.not_ok.find_each(&:check_status)
```

Comment:
The status check needs to happen before the `index_sitemaps` call. This ensures that all available domains will have their sitemaps checked. A spec should be added to ensure that this happens in the correct order.